kcidb/monitor/spool/__init__.py

```diff
@@ -99,27 +99,43 @@ def post(self, notification, timestamp=None):
                             datetime.datetime.now(datetime.timezone.utc).
 
         Returns:
-            True, if the notification was posted onto the spool,
-            False, if not (it was already there).
+            True, if the new notification was posted onto the spool,
+            or the existing, "unpicked" notification was updated,
+            False, if not (it existed and was "picked" at the moment).
         """
         assert isinstance(notification, Notification)
         assert timestamp is None or \
-            isinstance(timestamp, datetime.datetime) and timestamp.tzinfo
+               isinstance(timestamp, datetime.datetime) and timestamp.tzinfo
         if timestamp is None:
             timestamp = datetime.datetime.now(datetime.timezone.utc)
 
-        try:
-            self._get_doc(notification.id).create(dict(
-                created_at=timestamp,
-                # Set to a definitely timed-out time, free for picking
-                picked_until=datetime.datetime.min,
-                message=notification.render().as_string(
-                    policy=email.policy.SMTPUTF8
-                ),
-            ))
-        except Conflict:
-            return False
-        return True
+        @firestore.transactional
+        def create_or_update(transaction):
+            message = notification.render(). \
+                as_string(policy=email.policy.SMTPUTF8)
```

Comment:
I'm talking about this call of `notification.render()`. You do it here first, and then again when creating the document.

Reply:
Ahhh, I have seen it, Sorry 😢 about that.
I did not see it earlier

Reply:
No problem, that's what reviews are for :smile: